packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import uniqueid from 'lodash.uniqueid';
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+    * Whether or not the Switch is enabled ("ON").
+    */
+  isOn: PropTypes.bool,
+
+  /**
+    * If true, the user won't be able to toggle the switch.
+    */
+  isDisabled: PropTypes.bool,
+
+  /**
+    * Text of the label.
+    */
+  labelText: PropTypes.node.isRequired,
+
+  /**
+    * Callback fired when the state is changed.
+    */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  isOn: false,
+  isDisabled: false,
+  onChange: undefined,
+};
+
+const Switch = ({
```

Comment:
Is this meant to be an uncontrolled component? If not, then I'd expect us to maintain internal state of whether Switch is enabled/disabled and flip the toggle ourselves.

Reply:
I am not sure of about maintaining internal state to enable / disable Switch. but with current implementation user needs to handle this in onChange() callback provided by switch component. 

would like to if there are any scenarios which will help by keeping state variable for switch.

Reply:
I mean.. if it will need to be a controlled component, then that might need to be done, imo. According to the current implementation, it looks like one cannot flip the toggle if `onChange` prop is not supplied and if it is the expected behaviour, then that's fine.